### PR TITLE
Route summary

### PR DESCRIPTION
**Pull request by fernandogelin**

### CODE REVIEW

File: app/routes/erddap/buoy.js [hunk 1]
```diff
@@ -26,44 +33,59 @@ const { getMultiBuoyGeoJsonData, getBuoysCoordinates } = require('../../clients/
...
 
-  return Promise.all(getMultiBuoyGeoJsonData(payload)).then(
-    (response) => {
-      console.log(response);
-      const data = response.map((datum) => {
-        return datum.data.features.map((feature) => {
+  return Promise.all(getMultiBuoyGeoJsonData(payload)).then((response) => {
+    const data = response.map((datum) => {
+      if (datum.hasOwnProperty('data')) {
+        let processed = datum.data.features.map((feature) => {
```
Comment (mcmcgrath13) [inline app/routes/erddap/buoy.js:70]: should this data re-mapping be moved into the `getMultiBuoyGeoJSONData`?  That way if we use it outside of `/query` there's consistent keys?

File: app/routes/erddap/buoy.js [hunk 2]
```diff
@@ -80,26 +102,94 @@ router.get('/buoy', (req, res) => {
...
-          longitude: buoy.data.features[0].geometry.coordinates[0],
-          buoyId: buoy.data.features[0].properties.station_name
-        };
-      });
-      res.send(data);
-    }
-  );
-  
+  return Promise.all(getBuoysCoordinates({ ids })).then((response) => {
+    const data = response.map((buoy) => {
```
Comment (mcmcgrath13) [inline app/routes/erddap/buoy.js:113]: same question as above - where should the division of logic line be?

File: app/routes/erddap/buoy.js [hunk 3]
```diff
@@ -80,26 +102,94 @@ router.get('/buoy', (req, res) => {
...
+      (response) => {
+        const data = response.map((datum) => {
+          console.log(datum);
+          return datum.data?.features.map((feature) => {
+            return feature.properties;
+          });
+        });
+
+        const rollupObject = {};
+        buoys.variables.forEach(v => {
```
Comment (mcmcgrath13) [inline app/routes/erddap/buoy.js:166]: what does this do?

File: app/routes/erddap/buoy.js [hunk 4]
```diff
@@ -80,26 +102,94 @@ router.get('/buoy', (req, res) => {
...
+          rollupObject[v] = op.valid(v);
+        });
+
+        const processed = data.map((d) => {
+          let dt = aq.from(d)
+            .derive({
+              dt_ym: (d) =>
+                op.datetime(op.year(d.time), op.month(d.time)),
+              station_id: (d) => d.station_name
```
Comment (mcmcgrath13) [inline app/routes/erddap/buoy.js:175]: might be more efficient to do the `station_name` => `station_id` rename with a `select`

File: app/routes/erddap/buoy.js [hunk 5]
```diff
@@ -80,26 +102,94 @@ router.get('/buoy', (req, res) => {
...
+            })
+            .groupby('station_id', 'dt_ym')
+            .rollup(rollupObject)
+            .objects();
+          return dt;
+        });
+
+        const final = processed
+          .reduce((a, b) => a.concat(b))
```
Comment (mcmcgrath13) [inline app/routes/erddap/buoy.js:184]: should this have a default/start value of `[]`?

File: app/routes/erddap/utils.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+const ids = [
```
Comment (mcmcgrath13) [inline app/routes/erddap/utils.js:1]: does this really need to be hardcoded? or can we derive it from the available data in the dataset?

File: app/routes/erddap/utils.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  'bid15',
+  'bid16',
+  'bid17'
+];
+
+const variables = [
+  'O2PercentSurface',
+  'depth',
+  'SalinityBottom',
+  'pHBottom',
```
Comment (mcmcgrath13) [inline app/routes/erddap/utils.js:21]: same question as above

File: app/routes/erddap/utils.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  'FSpercentSurface',
+  'WaterTempBottom',
+  'O2Surface',
+  'WaterTempSurface',
+  'SalinitySurface'
+];
+
+const stationMap = {
+  bid2: 'N. Prudence',
+  bid3: 'Conimicut Pt',
```
Comment (mcmcgrath13) [inline app/routes/erddap/utils.js:37]: should this be on erddap somehow?

File: app/swaggerDef.js [hunk 9]
```diff
@@ -6,13 +6,13 @@ const host = `${process.env.HOST}:${process.env.PORT}`;
 module.exports = {
   definition: {
     info: {
-    // API informations (required)
+      // API informations (required)
       title: 'ERDDAP Proxy API', // Title (required)
```
Comment (mcmcgrath13) [inline app/swaggerDef.js:10]: RIDDC API for the title?

File: test/erddap_test.js [hunk 10]
```diff
@@ -1,25 +1,43 @@
...
+const app = require('@/server');
 
 chai.use(chaiHttp);
 chai.should();
+// const expect = chai.expect; 
 
 describe('Buoy', () => {
-  describe('GET /', () => {
-    // Test to get all students record
+  describe('GET /erddap/buoy', () => {
```
Comment (mcmcgrath13) [inline test/erddap_test.js:11]: should be `/erddap/buoy/query`?

File: app/routes/erddap/buoy.js (hunk 1 above)
Comment (fernandogelin) [reply app/routes/erddap/buoy.js:70]: good point

File: app/routes/erddap/buoy.js (hunk 3 above)
Comment (fernandogelin) [reply app/routes/erddap/buoy.js:166]: it creates the rollup object with all variables in it

Comment (mcmcgrath13) [reply app/routes/erddap/buoy.js:166]: still confused

File: test/erddap_test.js (hunk 10 above)
Comment (fernandogelin) [reply test/erddap_test.js:11]: it should, but I actually think we should just remove the `query` word from the route

File: app/routes/erddap/utils.js (hunk 8 above)
Comment (fernandogelin) [reply app/routes/erddap/utils.js:37]: potentially

File: test/erddap_test.js (hunk 10 above)
Comment (mcmcgrath13) [reply test/erddap_test.js:11]: why? I think the query makes a more idiomatic api

Comment (fernandogelin) [reply test/erddap_test.js:11]: `?` means `query` already, it's just repetitive and makes the url longer.

Comment (mcmcgrath13) [reply test/erddap_test.js:11]: but `/?kdljfaoweifa;soefh` looks funny (and I think I tried it and it didn't work)

Comment (fernandogelin) [reply test/erddap_test.js:11]: I'm confused, what did you try?
It should be just:

`/erddap/buoy?ids=1,2&variables=foo,bar"

The things after the `?` are query parameters.

Comment (mcmcgrath13) [reply test/erddap_test.js:11]: but because everything is mounted under `/buoy`, `/` refers to `/buoy/`

**Pull request merged**